packages/ui-react-utils/src/getElementType.ts

```diff
@@ -80,7 +75,7 @@ function getElementType<T extends ComponentWithAsProp>(
     return 'button'
   }
 
-  return Component.defaultProps.as || 'span'
+  return Component.defaultProps?.as ?? 'span'
```

Comment:
just FYI this refactor makes it behave slightly differently: 

```
class Component {
  static defaultProps = {as: ''}
}
const a = Component.defaultProps.as || 'span'
console.log("AA", a) // logs "AA span"
const b = Component.defaultProps?.as ?? 'span'
console.log("BB:", b) // logs "BB"
```
But I dont think this is an issue here